kubernetes/kubernetes.go

```diff
@@ -208,7 +214,28 @@ func CollectKubeMetrics(config KubeAgentConfig) {
 
 }
 
-func performConnectionChecks(ka *KubeAgentConfig) error {
+// isCustomS3UploadEnvsSet checks to see if the agent has a custom S3 location and S3 region to upload to
+// if both these variables are not set, default upload to Apptio S3
+func isCustomS3UploadEnvsSet(ka *KubeAgentConfig) bool {
+	if ka.CustomS3Region == "" && ka.CustomS3UploadBucket == "" {
+		return false
+	}
+	if ka.CustomS3UploadBucket == "" || ka.CustomS3Region == "" {
+		log.Warnf("Detected only one of the two required environment variables to run in custom S3 upload"+
```

Comment:
This seems like an invalid configuration. Would a fatal log be more appropriate here?

Reply:
Sure, will update